docs/internals/release-process.txt

```diff
@@ -37,39 +35,84 @@ security purposes, please see :doc:`our security policies <security>`.
-    feature is deprecated in version ``A.B``, it will continue to work in versions
-    ``A.B`` and  ``A.B+1`` but raise warnings. It will be removed in version
-    ``A.B+2``.
+    These releases will be 100% compatible with the associated feature release,
+    unless this is impossible for security reasons or to prevent data loss.
+    So the answer to "should I upgrade to the latest patch release?" will always
+    be "yes."
+
+  Long-term support release
+    Certain feature releases will be designated as "Long-term support" (LTS)
+    releases. LTS releases will get security and data loss fixes applied for a
+    guaranteed period of time, typically three years.
 
-    So, for example, if we decided to start the deprecation of a function in
-    Django 1.7:
+    See `the download page`_ for the releases that have been designated for
+    long-term support.
 
-    * Django 1.7 will contain a backwards-compatible replica of the function which
-      will raise a ``RemovedInDjango19Warning``. This warning is silent by
-      default; you can turn on display of these warnings with the ``-Wd`` option
-      of Python.
+    .. _the download page: https://www.djangoproject.com/download/
 
-    * Django 1.8 will still contain the backwards-compatible replica. This
-      warning becomes *loud* by default, and will likely be quite annoying.
+Release cadence
+===============
 
-    * Django 1.9 will remove the feature outright.
+Starting with Django 3.0, version numbers will use a loose form of `semantic
+versioning <http://semver.org/>`_ such that each version following an LTS will
+bump to the next "dot zero" version. For example: 3.0, 3.1, 3.2 (LTS), 4.0,
+4.1, 4.2 (LTS), etc.
 
-  Minor release
-    Minor releases (A.B.C, etc.) will be issued as needed, often to fix security
-    issues.
+SemVer makes it easier to see at a glance how compatible releases are with each
+other. It also helps to anticipate when compatibility shims will be removed.
+It's not a pure form of SemVer as each feature release will continue to have a
+few documented backwards incompatibilities where a deprecation path isn't
+possible or not worth the cost. Also, deprecations started in the version
+preceding an LTS (X.1) will be dropped in a non-dot-zero release to accommodate
+our policy of keeping deprecation shims for at least two releases. Read on to
+the next section for an example.
 
-    These releases will be 100% compatible with the associated major release,
-    unless this is impossible for security reasons or to prevent data loss.
-    So the answer to "should I upgrade to the latest minor release?" will always
-    be "yes."
+.. _internal-release-deprecation-policy:
+
+Deprecation policy
+==================
+
+A feature release may deprecate certain features from previous releases. If a
+feature is deprecated in version ``A.B``, it will continue to work in versions
+``A.B`` and  ``A.B+1`` but raise warnings. It will usually be removed in
```

Comment:
Maybe it would be clearer here to just talk about "feature release X" and "X+1" and "X+2"? Using `A.B` is intended to be more clear, but I think it may end up being less clear because in practice `A.B+1` will often be `A+1.0` instead, etc.